src/Engine/RadiumEngine.hpp

```diff
@@ -99,7 +106,18 @@ class RA_ENGINE_API RadiumEngine {
     const std::vector<std::shared_ptr<Asset::FileLoaderInterface>>& getFileLoaders() const;
 
   private:
-    std::map<std::string, std::shared_ptr<System>> m_systems;
+    using priority = int;
+    using SystemKey = std::pair<priority, std::string>;
+    using SystemContainer = std::map<SystemKey, std::shared_ptr<System>, std::greater<SystemKey>>;

```

Comment:
Then when going through this map, `System` with the same priority will be processed by reversed alphabetical order right?

Reply:
Yes, as it was before introducing the priority.
But for me this is implementation dependent and I don't think it is a good idea to explain it (we just say random), otherwise users may be tempted to alter the system name to change the order, instead of using the priority.

Reply:
Ok (though it was the other way around due to standard string comparison being lower, not greater ;) )

Reply:
You see, we don't want to explain on this ;)